### PR TITLE
9488 adc filtering to test

### PR DESCRIPTION
**Pull request by codyseibert**

### CODE REVIEW

File: web-client/src/presenter/computeds/formattedMessages.js [hunk 1]
```diff
@@ -21,12 +22,62 @@ export const formattedMessages = (get, applicationContext) => {
...
+      completedByFilter ? message.completedBy === completedByFilter : true,
+    );
+  }
+
+  const caseStatuses = uniq(map(filteredMessages, 'caseStatus'));
+  const toUsers = uniq(map(filteredMessages, 'to'));
+  const fromUsers = uniq(map(filteredMessages, 'from'));
+  const fromSections = uniq(map(filteredMessages, 'fromSection'));
+  const toSections = uniq(map(filteredMessages, 'toSection'));
+
```
Comment (jamesobrooks) [inline web-client/src/presenter/computeds/formattedMessages.js]: ⛏️  I would remove this blank line.

File: web-client/src/presenter/computeds/formattedMessages.test.js [hunk 2]
```diff
@@ -146,4 +150,338 @@ describe('formattedMessages', () => {
...
+          toSection: 'petitionsclerk',
+        }),
+      ]),
+    );
+    expect(result).toMatchObject({
+      completedByUsers: expect.arrayContaining(['bob']),
+    });
+  });
+
+  it('the return the messages unfiltered if no filters are set', () => {
```
Comment (jamesobrooks) [inline web-client/src/presenter/computeds/formattedMessages.test.js:330]:
```suggestion
  it('should return the messages unfiltered if no filters are set', () => {
```

File: web-client/src/presenter/computeds/formattedMessages.test.js [hunk 3]
```diff
@@ -146,4 +150,338 @@ describe('formattedMessages', () => {
...
+          message: 'This is a test message',
+          to: 'petitionsclerk',
+          toSection: 'petitionsclerk',
+        }),
+      ]),
+    );
+    expect(result.messages.length).toEqual(2);
+  });
+
+  it('the return the completed messaged unfiltered if no filters are set', () => {
```
Comment (jamesobrooks) [inline web-client/src/presenter/computeds/formattedMessages.test.js:368]:
```suggestion
  it('should return the completed messaged unfiltered if no filters are set', () => {
```

**Review by jamesobrooks — COMMENTED** [review]
I'm not sure we should do the filtering in `formattedMessages`. We (@Absolutestunna and @shumway-tm I believe) pulled sorting out of there because it was doing too much. While filtering does seem more related to the work that `formattedMessages` is intended to do than sorting is, I think it would be better to have a separate computed and sequence similar to sorting.

File: web-client/src/presenter/computeds/formattedMessages.test.js [hunk 4]
```diff
@@ -146,4 +150,342 @@ describe('formattedMessages', () => {
...
   });
+
+  it('the filter dropdown values should be set correctly for the messages data', () => {
+    const result = runCompute(formattedMessages, {
+      state: {
+        messageBoxToDisplay: {
+          box: 'outbox',
+        },
+        messages: [
+          {
```
Comment (rosiehsmith) [inline web-client/src/presenter/computeds/formattedMessages.test.js:165]: 💭  With all these messages in tests, maybe it'd be worth adding a `mockMessages` file like we have with `mockCase.js`

**Review by rosiehsmith — APPROVED** [review]

File: web-client/src/presenter/computeds/formattedMessages.js [hunk 5]
```diff
@@ -1,83 +1,61 @@
...
-
   const hasMessages = messages.length > 0;
 
-  return {
-    caseStatuses,
-    completedByUsers,
-    completedMessages: filteredCompletedMessages,
-    fromSections,
-    fromUsers,
+  let showFilters = role === USER_ROLES.adc;
```
Comment (jamesobrooks) [inline web-client/src/presenter/computeds/formattedMessages.js:27]: :pick: I might name this one `isFilterableAndSortable` or something else to indicate that it affects both things (notably line 56).

**Review by jamesobrooks — APPROVED** [review]

File: web-client/src/presenter/computeds/formattedMessages.js [hunk 6]
```diff
@@ -1,32 +1,59 @@
...
+    });
+
+    sharedComputedResult = {
+      ...sharedComputedResult,
+      ...messageFilterResults.filterValues,
+      ...completedMessageFilterResults.filterValues,
+      completedMessages:
+        completedMessageFilterResults.filteredCompletedMessages,
+      messages: messageFilterResults.filteredMessages,
+      showFilters: true,
```
Comment (rachelschneiderman) [inline web-client/src/presenter/computeds/formattedMessages.js]: Show filters would already be true right?

File: web-client/src/presenter/computeds/formattedMessages.test.js [hunk 7]
```diff
@@ -146,4 +154,240 @@ describe('formattedMessages', () => {
       messages: [],
     });
   });
+
+  it('the filter dropdown values should be set correctly for the messages data', () => {
```
Comment (rachelschneiderman) [inline web-client/src/presenter/computeds/formattedMessages.test.js]: Nit: What does correctly mean for this test? I don't see filterValues in the expectation at all. Is there a way the test name can be made more descriptive so if/when it fails it's a little more indicative of what we are expecting the behavior to be?

File: web-client/src/presenter/utilities/processFormattedMessages.test.js [hunk 8]
```diff
@@ -451,4 +454,114 @@ describe('processFormattedMessages', () => {
       ]);
     });
   });
+
+  describe('applyFiltersToCompletedMessages', () => {
+    it('returns the filteredValues and expected filtered completed messages', () => {
```
Comment (rachelschneiderman) [inline web-client/src/presenter/utilities/processFormattedMessages.test.js]: Nit: Could we add to the test description that we are filtering by completed user Rick?

File: web-client/src/presenter/utilities/processFormattedMessages.test.js [hunk 9]
```diff
@@ -451,4 +454,114 @@ describe('processFormattedMessages', () => {
...
+        },
+      });
+
+      expect(result.filteredCompletedMessages.length).toEqual(1);
+      expect(result.filterValues.completedByUsers).toEqual(['rick']);
+    });
+  });
+
+  describe('applyFiltersToMessages', () => {
+    it('returns the expected filterValues and filtered messages', () => {
```
Comment (rachelschneiderman) [inline web-client/src/presenter/utilities/processFormattedMessages.test.js]: Nit: Same here, what is the test doing and/or why do we expect the result?

File: web-client/src/presenter/computeds/formattedMessages.js (hunk 6 above)
Comment (codyseibert) [reply web-client/src/presenter/computeds/formattedMessages.js]: yeah good point

**Pull request closed without merging**